newIDE/app/src/UI/Reponsive/ResponsiveWindowMeasurer.js

```diff
@@ -7,12 +7,30 @@ type Props = {|
   children: (width: WidthType) => React.Node,
 |};
 
+const useForceUpdate = () => {
+  const [value, setValue] = React.useState(true);
+  return () => setValue(value => !value);
+};
+
 /**
  * Pass the proper size to the children according to the window size.
  * This consider a window to be "small" if *both* the width and height
  * are small.
  */
 export const ResponsiveWindowMeasurer = ({ children }: Props) => {
+  const forceUpdate = useForceUpdate();
+
+  React.useEffect(() => {
+    // Use timeouts to only rerender when the resize is finished.
+    let timeout;
+    const listener = window.addEventListener('resize', () => {
+      clearTimeout(timeout);
+      timeout = setTimeout(forceUpdate, 200);
+    });
+
+    return () => window.removeEventListener('resize', listener);
```

Comment:
You should probably clear the timeout here too, otherwise if I resize the window and just after the component is destroyed (because a dialog was closed/something else happened), the timeout will trigger and call forceUpdate on a component that was unmounted - this kind of leak can be very risky.

Would be worth also comparing this implementation with other deboucing/throttling implementation with React useEffect - these hooks are easy to get wrong.

I'm also a bit concerned about having a lot of "resize" event listener. Would you be able to launch this and use the Chrome debugger to see how many resize listeners are added? You can find these in:
![image](https://user-images.githubusercontent.com/1280130/96229432-cdf39f00-0f8e-11eb-8e9d-fca2bf9ca451.png)

Should be no more than a few according to what is on screen.